geonode/documents/api/views.py

```diff
@@ -119,23 +121,28 @@ def perform_create(self, serializer):
                 "extension": extension,
                 "resource_type": "document",
             }
-            if file:
-                manager = StorageManager(remote_files={"base_file": file})
-                manager.clone_remote_files()
-                payload["files"] = [manager.get_retrieved_paths().get("base_file")]
             if doc_url:
                 payload["doc_url"] = doc_url
                 payload["sourcetype"] = enumerations.SOURCE_TYPE_REMOTE
 
             resource = serializer.save(**payload)
 
+            if file:
+                manager = StorageManager(remote_files={"base_file": file})
+                manager.clone_remote_files()
+                create_asset_and_link(
```

Comment:
Instead of cloning the remote files and then copy them again in the `settings.ASSET_ROOT`, whey dont we improve the `clone_remote_files` by giving a destination path? In this way we can avoid to perform 3 operations on the file.
In geonode deployment with a huge quanity of document, this file system work can be an issue

Reply:
keeping the current behaviour so far